service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/pis/SinglePaymentSpiImpl.java

```diff
@@ -37,18 +37,18 @@ public SinglePaymentSpiImpl(TestDataService testDataService) {
     if (!psuId.isPresent()) {
       throw new RestException(HttpStatus.UNAUTHORIZED, "PSU not found");
     }
-    boolean isBlocked = testDataService.isBlockedPsu(psuId.get());
 
-    if (isBlocked) {
-      return SpiResponse.<SpiSinglePaymentInitiationResponse>builder()
-          .fail(SpiResponseStatus.TECHNICAL_FAILURE);
+    if (testDataService.isBlockedPsu(psuId.get())) {
+      // We need to throw here instead of returning a SpiResponse.failed() because if we did,
```

Comment:
Is this comment needed? I would delete it because MessageErrorCode.Service_blocked looks like the correct way

Reply:
I'm moving this to the commit MSG.